coops-ui/src/views/StandaloneDirectorsFiling.vue

```diff
@@ -0,0 +1,301 @@
+      }
+      const business = {
+        business: {
+          foundingDate: this.entityFoundingDate,
+          identifier: this.entityIncNo,
+          legalName: this.entityName
+        }
+      }
+
+      if (this.isDataChanged('OTCDR')) {
+        changeOfDirectors = {
+          changeOfDirectors: {
+            certifiedBy: 'Full Name',
+            email: 'no_one@never.get',
+            directors: this.$refs.directorsList.getAllDirectors()
+          }
+        }
+      }
+
+      const filingData = {
+        filing: Object.assign(
+          {},
+          header,
+          business,
+          changeOfDirectors
+        )
+      }
+
+      // Temp fix Added to remove cors error while invoking mock endpoint.
+      // To be removed when the API is in place
+      let config = {
+        headers: {
+          'Content-Type': 'text/plain'
+        }
+      }
+
+      axios.post(this.corpNum + '/filings', filingData, config).then(res => {
+        let payRequestId:String = res.data.filing.header.paymentToken
+        payRequestId = '189'
+        let returnURL = window.location.origin + '/standalone-directors?pay_id=' + payRequestId
+        let authStub:string = this.authURL
+        if (!(authStub.endsWith('/'))) {
+          authStub = authStub + '/'
+        }
+        let payURL = authStub + 'makepayment/' + payRequestId + '/' + encodeURIComponent(returnURL)
+        window.location.href = payURL
+      }).catch((error) => {
+        // TODO : To Decide how and where to display the error message from API
+        console.log(error)
+        this.dialog = true
+      })
+    },
+
+    toggleFiling (setting, filing) {
+      let added = false
+      for (let i = 0; i < this.filingData.length; i++) {
+        if (this.filingData[i].filingTypeCode === filing) {
+          if (setting === 'add') {
+            added = true
+          } else {
+            this.filingData.splice(i, 1)
+          }
+          break
+        }
+      }
+      if (setting === 'add' && !added) {
+        this.filingData.push({ filingTypeCode: filing, entityType: 'CP' })
+      }
+    },
+
+    isDataChanged (key) {
+      return this.filingData.find(o => o.filingTypeCode === key)
+    },
+
+    navigateToDashboard () {
+      this.dialog = false
+      this.$router.push('/')
+    },
+
+    setValidateFlag () {
+      // compute the AR page's valid state
+      this.setValidated(this.agmDateValid && this.addressesFormValid && this.directorFormValid)
+    }
+  },
+
+  watch: {
+
+    directorsChange: function (val) {
+      // when directors change, update filing data
+      console.log('Standalone Directors, directorsChange =', val)
+      if (val) {
+        this.toggleFiling('add', 'OTCDR')
+      } else {
+        this.toggleFiling('remove', 'OTCDR')
+      }
+    },
+
+    directorFormValid (val) {
+      this.setValidateFlag()
+    },
+
+    filingData: function (val) {
+      console.log('StandaloneDirectorsFiling, filingData =', val)
```

Comment:
Yet again something to ponder - we don't want console.log code going to production. I think there are ways to strip it out, but if we do so in this instance would we still have the watcher? Just thinking about keeping it as simple as possible / how valuable is this code (yes, it's probably a carryover from AnnualReport).